src/main/java/org/sasanlabs/service/vulnerability/csrf/CSRFVulnerability.java

```diff
@@ -0,0 +1,249 @@
+public class CSRFVulnerability {
+
+    private static final String USR_LOGIN = "id";
+    private static final String USR_PASS = "pass";
+    private static final String USR_NEWPASS = "newpass";
+    private static final String TOKEN_PARAM = "rtkparam";
+    private static final String FIXED_TOKEN = "7877864849";
+
+    private static final String LOGIN = "mylogin";
+    private String currentPassword = "mypassword";
+
+    private Boolean userIsLoggedIn = false;
+    private String secretToken = "";
+
+    private static final SecureRandom rand = new SecureRandom();
+
+    private String checkInputs(
+            String userId, String userPassword, String newUserPassword, Boolean isSecure) {
+        BooleanSupplier validatorId = () -> StringUtils.isNotBlank(userId + userPassword);
+        BooleanSupplier validatorNP = () -> StringUtils.isNotBlank(newUserPassword);
+        String result;
+        if (Boolean.TRUE.equals(validatorId.getAsBoolean())) {
+            if (userId.equals(LOGIN) && userPassword.equals(currentPassword)) {
+                userIsLoggedIn = true;
+                result = "<div id=\"login\">[Logged in]</div> Hello " + userId + ".";
+            } else {
+                userIsLoggedIn = false;
+                result = "<div id=\"logout\">[Not Logged in]</div> Wrong credentials.";
+            }
+        } else if (Boolean.TRUE.equals(validatorNP.getAsBoolean())) {
+            if (userIsLoggedIn && isSecure) {
+                currentPassword = newUserPassword;
+                result = "<div id=\"login\">[Logged in]</div> Password changed correctly.";
+            } else {
+                result =
+                        (userIsLoggedIn
+                                ? "Password change refused."
+                                : "<div id=\"logout\">[Not Logged in]</div> You are not logged in. Login to change password.");
+            }
+        } else {
+            result = "BAD INPUT !";
+        }
+        return result;
+    }
+
+    private String checkInputsToken(
+            String usrLiveToken,
+            String usrGenToken,
+            String usrId,
+            String usrPass,
+            String usrNewPass) {
+        BooleanSupplier validator = () -> StringUtils.isNotBlank(usrLiveToken);
+        String rtn;
+        if (validator.getAsBoolean() && usrLiveToken.equals(usrGenToken)) {
+            rtn = checkInputs(usrId, usrPass, usrNewPass, true);
+        } else {
+            rtn = checkInputs(usrId, usrPass, usrNewPass, false);
+        }
+        return rtn;
+    }
+
+    @AttackVector(
+            vulnerabilityExposed = VulnerabilityType.CSRF,
+            description = "CSRF_VULNERABILITY_USER_INPUT1",
+            payload = "CSRF_PAYLOAD_LEVEL_1")
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_1,
+            htmlTemplate = "LEVEL_1/CSRFVulnerability")
+    public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerableCSRFLvl1(
```

Comment:
I would suggest that create another Api for just login and let this one for only change password.

So one method per vulnerability will be for login and that will be first called by the user. once login is completed, then show change password UI screen and let user change the password.